### PR TITLE
remove gogoproto replace

### PR DESCRIPTION
**Pull request by tac0turtle**
Closes: #XXX

## What is the purpose of the change

removes grpc replace to unblock some users 

## Brief Changelog

*(for example:)*
 
  - *The metadata is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *Daemons retrieve the RPC data from the blob cache*

## Testing and Verifying

*(Please pick one of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added unit test that validates ...*
  - *Added integration tests for end-to-end deployment with ...*
  - *Extended integration test for ...*
  - *Manually verified the change by ...*

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: baseapp/grpcrouter.go [hunk 1]
```diff
@@ -83,21 +84,18 @@ func (qrt *GRPCQueryRouter) RegisterService(sd *grpc.ServiceDesc, handler interf
 			// call the method handler from the service description with the handler object,
 			// a wrapped sdk.Context with proto-unmarshaled data from the ABCI request data
 			res, err := methodHandler(handler, sdk.WrapSDKContext(ctx), func(i interface{}) error {
-				err := protoCodec.Unmarshal(req.Data, i)
```
Comment (mattverse) [inline baseapp/grpcrouter.go:86]: is `protoCodec` still being used after this change?

**Review by mattverse — COMMENTED** [review]
utACK, is this something that is already implemented in upstream sdk and being used in prod?

File: baseapp/grpcrouter.go (hunk 1 above)
Comment (tac0turtle) [reply baseapp/grpcrouter.go:86]: yea, i am runnning it on osmosis mainnet as well

Comment (mattverse) [reply baseapp/grpcrouter.go:86]: sweet, happy to merge this upon seeing tests pass

**Comment by p0mvn** [comment]
Note to reviewer: backport to https://github.com/osmosis-labs/cosmos-sdk/tree/v16.x

**Pull request closed without merging**